internal/ast/utilities.go

```diff
@@ -903,6 +904,11 @@ func SetParentInChildren(node *Node) {
 	fn(node)
 }
 
+// This should never be called outside the parser
+func SetImportsOfSourceFile(node *SourceFile, imports []*LiteralLikeNode) {
```

Comment:
Should this just be a method? Or just provide imports via `NewSourceFile` like we do the rest? (Probably can't because it's a second pass ☹️)

Reply:
Yeah, can't be in the constructor because of the late-calculation thing, and I'm not sure something like this, which aught not be used without very good reason, should be on the method completion list. I'd rather not put mutating methods on AST nodes.